src/controller.cpp

```diff
@@ -594,7 +594,7 @@ void Controller::mark_article_read(const std::string& guid, bool read)
 
 void Controller::mark_all_read(unsigned int pos)
 {
-	if (pos < feedcontainer.feeds.size()) {
+	if (pos < feedcontainer.feeds_size()) {
```

Comment:
This function might contain a race condition.

The validity of the `pos` value is checked here but the number of feeds might be changed before the two uses of `pos`:
- `const auto feed = feedcontainer.get_feed(pos);`
- `feedcontainer.mark_all_feed_items_read(pos);`

Reply:
Fixed by the last three commits I just pushed.

Reply:
Looks good, thanks for fixing!